Essentials/src/com/earth2me/essentials/commands/Commandtime.java

```diff
@@ -166,25 +130,29 @@ private String normalizeWorldName(World world) {
 
     @Override
     protected List<String> getTabCompleteOptions(Server server, CommandSource sender, String commandLabel, String[] args) {
-        final User user = ess.getUser(sender.getPlayer());
-
+        boolean isConsole = sender.isPlayer();
         if (args.length == 1) {
-            if (user == null || user.isAuthorized("essentials.time.set")) {
+            if (isConsole || sender.isAuthorized("essentials.time.set", ess)) {
                 return Lists.newArrayList("set", "add");
             } else {
                 return Collections.emptyList();
             }
-        } else if (args.length == 2 && args[0].equalsIgnoreCase("set")) {
-            return Lists.newArrayList("sunrise", "day", "morning", "noon", "afternoon", "sunset", "night", "midnight");
-            // TODO: handle tab completion for add
+        } else if (args.length == 2) {
+            if (args[0].equalsIgnoreCase("set")) {
+                return Lists.newArrayList("sunrise", "day", "morning", "noon", "afternoon", "sunset", "night", "midnight");
+            } else if (args[0].equalsIgnoreCase("add")) {
+                return Lists.newArrayList("500", "1000", "1500", "2000", "2500", "3000");
```

Comment:
Numbers are a bit weird to autocomplete (they are ordered in lexicographical order), so maybe leave out 500 and increment by 1000 or something, or just leave out completions entirely.